src/multiclient/multi_client_actor.cpp

```diff
@@ -0,0 +1,266 @@
+    std::string result_ls_array;
+    {
+      td::JsonBuilder builder;
+      auto arr = builder.enter_array();
+      arr << ls_json;
+      arr.leave();
+      result_ls_array = builder.string_builder().as_cslice().str();
+    }
+
+    auto conf = td::json_decode(global_config).move_as_ok();
+
+    std::string result_config_str;
+    {
+      td::JsonBuilder builder;
+      auto obj = builder.enter_object();
+      obj("dht", get_json_object_field(conf.get_object(), "dht", td::JsonValue::Type::Object).move_as_ok());
+      obj("@type", get_json_object_field(conf.get_object(), "@type", td::JsonValue::Type::String).move_as_ok());
+      obj("validator", get_json_object_field(conf.get_object(), "validator", td::JsonValue::Type::Object).move_as_ok());
+      obj("liteservers", td::JsonRaw(result_ls_array));
+      obj.leave();
+      result_config_str = builder.string_builder().as_cslice().str();
+    }
+
+    result.push_back(std::move(result_config_str));
+  }
+
+  return result;
+}
+
+static auto kRandomDevice = std::random_device();
+static auto kRandomEngine = std::default_random_engine(kRandomDevice());
+
+template <typename T>
+T get_random_index(T from, T to) {
+  std::uniform_int_distribution<T> distribution(from, to);
+  return distribution(kRandomEngine);
+}
+
+}  // namespace
+
+void MultiClientActor::start_up() {
+  CHECK(std::filesystem::exists(config_.global_config_path));
+
+  auto global_config = td::read_file_str(config_.global_config_path.string()).move_as_ok();
+  auto config_splitted_by_liteservers = split_global_config_by_liteservers(std::move(global_config));
+
+  CHECK(!config_splitted_by_liteservers.empty());
+
+  if (config_.key_store_root.has_value()) {
+    if (std::filesystem::exists(*config_.key_store_root)) {
+      if (config_.reset_key_store) {
+        std::filesystem::remove_all(*config_.key_store_root);
+        std::filesystem::create_directories(*config_.key_store_root);
+      }
+    }
+  }
+
+  LOG(INFO) << "starting " << config_splitted_by_liteservers.size() << " client workers";
+
+  for (size_t i = 0; i < config_splitted_by_liteservers.size(); i++) {
+    workers_.push_back(WorkerInfo{
+        .id = td::actor::create_actor<ClientWrapper>(
+            td::actor::ActorOptions().with_name("multiclient_worker_" + std::to_string(i)).with_poll(),
+            ClientConfig{
+                .global_config = config_splitted_by_liteservers[i],
+                .key_store = config_.key_store_root.has_value() ?
+                    std::make_optional<std::filesystem::path>(*config_.key_store_root / ("ls_" + std::to_string(i))) :
+                    std::nullopt,
+                .blockchain_name = config_.blockchain_name,
+            }
+        ),
+    });
+  }
+
+  alarm_timestamp() = td::Timestamp::in(1.0);
+  next_archival_check_ = td::Timestamp::in(2.0);
+}
+
+void MultiClientActor::alarm() {
+  static constexpr double kDefaultAlarmInterval = 1.0;
+  static constexpr double kCheckArchivalInterval = 10.0;
```

Comment:
liteservers usually don't change `is_archival` often, I think we can set interval to 10 minutes